### DISCUSSION TITLE
stm32duino / VL53L1 library example not working

### DISCUSSION THREAD

**Discussion by metanav** [stm32duino/Arduino_Core_STM32#2239] [Q&A] [answered]
What is the value of the argument (0x12) in the code below. Is this some register address value or the I2C device address? 
I am using an Arduino  Nicla Vision with the onboard VL53L1CBV0FY/1. The program gets stuck at this line.

https://github.com/stm32duino/VL53L1/blob/c0474f235d3c86fd762b088cba687325d5769594/examples/VL53L1_Sat_HelloWorld/VL53L1_Sat_HelloWorld.ino#L93

**Comment by fpistm** [comment]
Hi @metanav 
It seems is is the I2C address:
https://github.com/stm32duino/VL53L1/blob/c0474f235d3c86fd762b088cba687325d5769594/src/vl53l1_class.cpp#L469-L484

But I let @cparata confirmed.

**Comment by cparata** [comment, marked as answer]
Hello @metanav ,
yes it is the I2C address to be used. Could you try to use the default address (no parameter) or change the I2C address to 0x08 or 0x40? Let me know if in this way it works.
Best Regards,
Carlo
